PackageKit/backends/zypp/pk-backend-zypp.cpp

```diff
@@ -2079,18 +2082,46 @@ zypp_perform_execution (PkBackendJob *job, ZYpp::Ptr zypp, PerformType type, gbo
 				total_remove_bytes,
 				total_cached_bytes);
 
-		int64_t required_space_bytes = (total_download_bytes + total_install_bytes - total_remove_bytes);
-		// XXX: This assumes package downloads also end up in rootfs, and that
-		// installed files will all take up space in the rootfs only
-		int64_t free_space_bytes = get_free_disk_space("/");
-		int64_t remaining_space_bytes = free_space_bytes - required_space_bytes;
+		int64_t required_space_bytes_download = total_download_bytes;
+		int64_t required_space_bytes_installation = (type == UPGRADE) ? (total_install_bytes - total_remove_bytes)
```

Comment:
The minimum required space for the installation step should be the installation size of the biggest package. This should be the same for the UPGRADE case and for an ordinary (app) installation.